### ISSUE TITLE
Serialization does not work for the new GPU backend

### ISSUE THREAD

**Issue by rizar** [mila-udem/blocks#1060]
```
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/main_loop.py", line 197, in run
    reraise_as(e)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/utils/__init__.py", line 258, in reraise_as
    six.reraise(type(new_exc), new_exc, orig_exc_traceback)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/main_loop.py", line 183, in run
    while self._run_epoch():
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/main_loop.py", line 230, in _run_epoch
    self._run_extensions('before_epoch')
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/main_loop.py", line 263, in _run_extensions
    extension.dispatch(CallbackName(method_name), *args)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/extensions/__init__.py", line 340, in dispatch
    self.do(callback_invoked, *(from_main_loop + tuple(arguments)))
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/extensions/saveload.py", line 98, in do
    use_cpickle=self.use_cpickle)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 220, in secure_dump
    dump_function(object_, temp, **kwargs)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 432, in dump_and_add_to_dump
    protocol=protocol, **kwargs)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 193, in dump
    _taradd(_save_parameters, tar_file, '_parameters')
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 606, in _taradd
    func(temp_file)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 191, in _save_parameters
    type(array_), n)
  File "/u/bahdanau/Dist/fully-neural-lvsr/libs/blocks/blocks/serialization.py", line 579, in _mangle_parameter_name
    return '#{}.{}'.format(_ARRAY_TYPE_MAP[type_], name)
KeyError: "<type 'pygpu.gpuarray.GpuArray'>\n\nOriginal exception:\n\tKeyError: <type 'pygpu.gpuarray.GpuArray'>" 
```

**Comment by dwf** [comment]
@rizar is this closed by #1063 and/or #1069?

**Comment by dwf** [comment]
Skimming over #1069 I'm pretty sure this is fixed, feel free to reopen if not.
